pyproject.toml

```diff
@@ -33,3 +33,40 @@ exclude = '''
   | treeherder/changelog/migrations
 )/
 '''
+
+[tool.ruff]
+# Same as Black.
+line-length = 100
+
+# Assume Python 3.9
+target-version = "py39"
+
+# In addition to the standard set of exclusions, omit all tests, plus a specific file.
+extend-exclude = ["*/.*/",".*/","__pycache__","node_modules"]
+
+select = [
+  # pycodestyle
+  "E",
+  "W",
+  # pyflakes
+  "F",
+]
+
+ignore = [
+  # E203 whitespace before ':'
+  "E203",
+  # E231: missing whitespace after ','
+  "E231",
+  # E501: line too long
+  "E501"
+]
+
+[tool.ruff.per-file-ignores]
+# Ignore `module-import-not-at-top-of-file` rule of `pycodestyle`
+"treeherder/model/models.py" = ["E402"]
+
+# Ignore `not-is-test` rule of `pycodestyle`
+"treeherder/perf/sheriffing_criteria/criteria_tracking.py" = ["E714"]
```

Comment:
Ruff suggests

```diff
--- a/treeherder/perf/sheriffing_criteria/criteria_tracking.py
+++ b/treeherder/perf/sheriffing_criteria/criteria_tracking.py
@@ -128,9 +128,9 @@ class ConcurrencyStrategy:
 
         if not issubclass(self._pool_class, Pool):
             raise TypeError(f'Expected Pool (sub)class parameter. Got {self._pool_class} instead')
-        if not type(thread_wait) is timedelta:
+        if type(thread_wait) is not timedelta:
             raise TypeError('Expected timedelta parameter.')
-        if not type(check_interval) is timedelta:
+        if type(check_interval) is not timedelta:
             raise TypeError('Expected timedelta parameter.')
```

Reply:
Done in https://github.com/mozilla/treeherder/pull/7898/commits/bdcbae98e269b20edd3d0e0732bd53011aec04ac